Tests/SwiftDiagnosticsTest/DiagnosticTestingUtils.swift

```diff
@@ -0,0 +1,221 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+/// A typealias representing a location marker.
+///
+/// This string serves to pinpoint the exact location of a particular token in the Abstract Syntax Tree (AST).
```

Comment:
Some terminology: I try to use AST for the C++ AST tree that’s used in the compiler, which isn’t source-accurate (that’s the abstract part to me). swift-syntax works on SwiftSyntax trees or syntax trees.